botocore/exceptions.py

```diff
@@ -52,6 +53,16 @@ class EndpointConnectionError(BotoCoreError):
         'Could not connect to the endpoint URL: "{endpoint_url}"')
 
 
+class BadStatusLineError(ConnectionError):
+    fmt = (
+        'Bad status line occurred on endpoint URL: "{endpoint_url}". '
```

Comment:
Well then. I'll go back to your suggestion of "closed", but use a passive form.

    class ConnectionClosed(ConnectionError):
        fmt = 'Connection is closed before we received a valid response from endpoint URL: "{endpoint_url}".'